src/core/store/modules/metadata/metadata.reducer.ts

```diff
@@ -0,0 +1,55 @@
+import { isEqual } from 'lodash';
+import { createReducer } from '@reduxjs/toolkit';
+
+import { ENSAddressPair, OnChainMetaDataState } from '@types';
+
+import { OnChainMetaDataActions } from './metadata.actions';
+
+const { getABI, clearABI, getENS } = OnChainMetaDataActions;
+
+export const initialOnChainMetaDataState: OnChainMetaDataState = {
+  contractABI: undefined,
+  contractFunctions: undefined,
+  ENS: [],
+};
+
+export const onChainMetaDataReducer = createReducer(initialOnChainMetaDataState, (builder) => {
```

Comment:
`onchain` not `onChain` throughout codebase plz

Reply:
Done